app/scripts/directives/overview/virtualMachineRow.js

```diff
@@ -87,15 +161,94 @@
       );
     };
     row.canStartOvm = function () {
-      return !isOvmRunning();
+      return !row.isOvmRunning();
     };
     row.canStopOvm = function () {
-      return isOvmRunning();
+      return row.isOvmRunning();
     };
     row.canRestartOvm = function () {
-      return isOvmRunning() &&
-             row.apiObject._vm &&
-             _.get(row.apiObject, '_pod.status.phase') === 'Running';
+      return row.isOvmInRunningPhase();
+    };
+    row.isWindowsVM = function () {
+      // https://github.com/kubevirt/kubevirt/blob/576d232e3c181134e69719cdb2d624ac52e7eecb/docs/devel/guest-os-info.md
+      var ovm = row.apiObject;
+      var vm = row.apiObject._vm;
+      var os = _.get(vm, 'metadata.labels["kubevirt.io/os"]') || _.get(ovm, 'metadata.labels["kubevirt.io/os"]');
```

Comment:
Unless I've misread, you're checking the same label twice.

Reply:
It accesses two different objects - `vm` and `ovm`.
In normal case, the value should be same.
But if differs (misconfiguration, different flow) or the `ovm` object is missing (case of ephemeral machines), the value from `VirtualMachine` object takes precedence.